### PR TITLE
add Auth interface

### PR DESCRIPTION
**Pull request by ynewmann**

### CODE REVIEW

File: auth.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+type AccessToken []byte
+
+type Auth interface {
+
+	// GenerateToken generates a new token for some time with id and allowed routes
+	GenerateToken(context.Context, string, time.Duration, ...string) (AccessToken, error)
+
+	// VerifyAccess verifies token access
```
Comment (Wondertan) [inline auth.go:15]: Verifies access to what? ;)

File: auth.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	VerifyAccessTo(context.Context, string, AccessToken) error
```
Comment (Wondertan) [inline auth.go:16]: As a method user, I don't understand what `To` from the name points to

File: auth.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+	// List returns list of ids of all active tokens from store
+	List(context.Context) ([]string, error)
```
Comment (Wondertan) [inline auth.go:19]: There are `RevokeToken` and `GenerateToken`, why not `ListTokens` then?

**Comment by Wondertan** [comment]
@jenyasd209, It is not mandatory to fix those comments, as we have different priorities, but I would not say the API is finished.

**Comment by ynewmann** [comment]
Outdated.
Requires some reworks.

Related to https://github.com/proximax-storage/go-xpx-dfms/pull/649 and https://github.com/proximax-storage/go-xpx-dfms-api-http/pull/58

**Pull request closed without merging**